pickLayer/core/set_active_layer_tool.py

```diff
@@ -101,33 +98,63 @@ def _get_default_search_radius(self) -> float:
             * context.mapToPixel().mapUnitsPerPixel()
         )
 
+    def _get_distance_to_feature_on_layer(
+        self,
+        layer: QgsVectorLayer,
+        feature: QgsFeature,
+        origin_map_point: QgsPointXY,
+    ) -> float:
+        # for unknown reasons saving all geoms to variables avoids fatal exs
+        origin_layer_point = self.toLayerCoordinates(layer, origin_map_point)
+        origin_geom = QgsGeometry.fromPointXY(origin_layer_point)
+        feature_geom = feature.geometry()
+        closest_geom = feature_geom.nearestPoint(origin_geom)
+        closest_layer_point = closest_geom.asPoint()
+        closest_map_point = self.toMapCoordinates(layer, closest_layer_point)
+        return origin_map_point.distance(closest_map_point)
+
     def _choose_layer_from_identify_results(
-        self, results: QgsMapToolIdentify.IdentifyResult
+        self,
+        results: List[QgsMapToolIdentify.IdentifyResult],
+        origin_map_coordinates: QgsPointXY,
     ) -> Optional[QgsMapLayer]:
 
-        # Preserve IdentifyResult order
-        resulting_layers = [result.mLayer for result in results]
-        if len(results) == 0:
-            return None
-        else:
-            return self._choose_layer(resulting_layers)
-
-    def _choose_layer(self, layers: List[QgsMapLayer]) -> QgsMapLayer:
-        """
-        Picks one layer from input layers (may contain duplicates) in
-        following order: point, line, polygon, other layers
-        """
-        preferred_geometry_type_order = {
-            QgsWkbTypes.PointGeometry: 1,
+        type_preference = {
+            QgsWkbTypes.PointGeometry: 3,
             QgsWkbTypes.LineGeometry: 2,
-            QgsWkbTypes.PolygonGeometry: 3,
+            QgsWkbTypes.PolygonGeometry: 1,
         }
 
-        return sorted(
-            layers,
-            key=lambda layer: preferred_geometry_type_order.get(
-                layer.geometryType(), 99
-            )
-            if isinstance(layer, QgsVectorLayer)
-            else 99,
-        )[0]
+        best_match: Optional[QgsVectorLayer] = None
+        best_match_type_preference = 0
+        best_match_distance = 0.0
+
+        for result in results:
+            # sanity check even if identify type is vector layer, helps with type hints
```

Comment:
This comment is probably not needed? Or maybe fix sentence to "sanity check to help with type hinting"